tests/models/seggpt/test_modeling_seggpt.py

```diff
@@ -0,0 +1,383 @@
+        pixel_values = floats_tensor([self.batch_size, self.num_channels, self.image_size // 2, self.image_size])
+        prompt_pixel_values = floats_tensor(
+            [self.batch_size, self.num_channels, self.image_size // 2, self.image_size]
+        )
+        prompt_masks = floats_tensor([self.batch_size, self.num_channels, self.image_size // 2, self.image_size])
+
+        labels = None
+        if self.use_labels:
+            labels = floats_tensor([self.batch_size, self.num_channels, self.image_size // 2, self.image_size])
+
+        config = self.get_config()
+
+        return config, pixel_values, prompt_pixel_values, prompt_masks, labels
+
+    def get_config(self):
+        return SegGptConfig(
+            image_size=self.image_size,
+            patch_size=self.patch_size,
+            num_channels=self.num_channels,
+            hidden_size=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            hidden_act=self.hidden_act,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+            initializer_range=self.initializer_range,
+            mlp_ratio=self.mlp_ratio,
+            merge_index=self.merge_index,
+            out_indicies=self.out_indicies,
+            pretrain_image_size=self.pretrain_image_size,
+            decoder_hidden_size=self.decoder_hidden_size,
+        )
+
+    def create_and_check_model(self, config, pixel_values, prompt_pixel_values, prompt_masks, labels):
+        model = SegGptModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values, prompt_pixel_values, prompt_masks)
+        self.parent.assertEqual(
+            result.last_hidden_state.shape,
+            (
+                self.batch_size,
+                self.image_size // self.patch_size,
+                self.image_size // self.patch_size,
+                self.hidden_size,
+            ),
+        )
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        (
+            config,
+            pixel_values,
+            prompt_pixel_values,
+            prompt_masks,
+            labels,
+        ) = config_and_inputs
+        inputs_dict = {
+            "pixel_values": pixel_values,
+            "prompt_pixel_values": prompt_pixel_values,
+            "prompt_masks": prompt_masks,
+        }
+        return config, inputs_dict
+
+
+@require_torch
+class SegGptModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
+    """
+    Here we also overwrite some of the tests of test_modeling_common.py, as SegGpt does not use input_ids, inputs_embeds,
+    attention_mask and seq_length.
+    """
+
+    all_model_classes = (
+        (
+            SegGptModel,
+            SegGptForImageSegmentation,
+        )
+        if is_torch_available()
+        else ()
+    )
+    fx_compatible = False
```

Comment:
Has this been tested?

Reply:
+1

Reply:
What is fx_compatible?

Reply:
Set it to True and tested locally, but got `not supported yet`, didn't dig that much, but right now I'm not sure how I can add support for `SegGpt` for that